src/common/monitor/Monitor.cpp

```diff
@@ -5,16 +5,18 @@
 ///
 //===----------------------------------------------------------------------===//
 
+#include <common/debug/Log.h>
+#include <common/kafka/KafkaConfig.h>
 #include <common/monitor/Monitor.h>
 
-#include <common/debug/Log.h>
 //#undef TRC_MASK
 //#define TRC_MASK 0
 
 Monitor::Monitor(const std::string &broker, const std::string &topic_prefix,
                  const std::string &source_name) {
   source_name_ = source_name;
-  producer = std::make_shared<Producer>(broker, topic_prefix + "_monitor");
+  producer = std::make_shared<Producer>(broker, topic_prefix + "_monitor",
+      KafkaConfig::DefaultConfig);
```

Comment:
Will Monitor always use `KafkaConfig::DefaultConfig`?
Any Kafka connections that ignore settings provided in the config file may fail to authenticate to Kafka, if authentication is enforced.

(We also face this problem in the filewriter. Some Kafka settings like `message.max.bytes` feel quite topic-specific, but others like `security.protocol` must be used in all Kafka interactions. So it's a little difficult to store all settings in a single "flat" configuration).

Reply:
Just investigated: Monitor class is not used in ESS production code. Some uses may be in efu_legacy_modules repo but code in here is no longer supported. Code can be brought back to life but builds are not guaranteed to not be broken :-)